### PR TITLE
Add test cases of search expression

### PR DESCRIPTION
**Pull request by NicoYuan1986**
link: #23646

### CODE REVIEW

File: tests/python_client/testcases/test_search.py [hunk 1]
```diff
@@ -2667,6 +2667,41 @@ def test_search_expression_all_data_type(self, nb, nq, dim, auto_id, _async):
...
+        collection_w = self.init_collection_wrap(schema=default_schema)
+        collection_w = cf.insert_data(collection_w, is_all_data_type=True, insert_offset=-1500)[0]
+
+        # 2. create index and load
+        collection_w.create_index(field_name, default_index_params)
+        collection_w.load()
+
+        # 3. search
+        log.info("test_search_expression_different_data_type: Searching collection %s" % collection_w.name)
+        expression = "bool == True" if field == "bool" else f"{field} >= 1"
```
Comment (zhuwenxing) [inline tests/python_client/testcases/test_search.py:2690]: `f"{field} >= 1"` may not trigger the issue.
It was triggered due to being out of bounds. 
for example, the range of the int8 datatype is [-128, 127], so you need to set the expression as int8 > 128 or int8 < -129.

Comment (NicoYuan1986) [reply tests/python_client/testcases/test_search.py:2690]: Great idea. I'll modify the cases.

**Pull request merged**